Cesium3DTilesSelection/src/TilesetContentManager.cpp

```diff
@@ -1050,46 +1075,50 @@ bool TilesetContentManager::unloadTileContent(Tile& tile) {
   }
   tile.getMappedRasterTiles().clear();
 
-  // Unload the renderer resources and clear any raster overlay tiles. We can do
-  // this even if the tile can't be fully unloaded because this tile's geometry
-  // is being using by an async upsample operation (checked below).
-  switch (state) {
-  case TileLoadState::ContentLoaded:
-    unloadContentLoadedState(tile);
-    break;
-  case TileLoadState::Done:
-    unloadDoneState(tile);
-    break;
-  default:
-    break;
+  if (!tile.isEmptyContent() && !tile.isUnknownContent()) {
+    // Unload the renderer resources and clear any raster overlay tiles. We can
+    // do this even if the tile can't be fully unloaded because this tile's
+    // geometry is being using by an async upsample operation (checked below).
+    switch (state) {
+    case TileLoadState::ContentLoaded:
+      unloadContentLoadedState(tile);
+      break;
+    case TileLoadState::Done:
+      unloadDoneState(tile);
+      break;
+    default:
+      break;
+    }
   }
 
-  // Are any children currently being upsampled from this tile?
-  for (const Tile& child : tile.getChildren()) {
-    if (child.getState() == TileLoadState::ContentLoading &&
-        std::holds_alternative<CesiumGeometry::UpsampledQuadtreeNode>(
-            child.getTileID())) {
-      // Yes, a child is upsampling from this tile, so it may be using the
-      // tile's content from another thread via lambda capture. We can't unload
-      // it right now. So mark the tile as in the process of unloading and stop
-      // here.
-      tile.setState(TileLoadState::Unloading);
-      return false;
+  if (isReadyToUnload) {
+    // Make sure we unload all children
+    gsl::span<Cesium3DTilesSelection::Tile> children = tile.getChildren();
+    for (Tile& child : children) {
+      this->unloadTileContent(child);
     }
   }
 
   // If we make it this far, the tile's content will be fully unloaded.
   notifyTileUnloading(&tile);
   content.setContentKind(TileUnknownContent{});
-  tile.setState(TileLoadState::Unloaded);
+  if (isExternalContent) {
+    // We don't want to set external tilesets as Unloaded quite yet, because
+    // then they might get reloaded before we've had a chance to clear their
+    // children and cause an error. They'll get their children cleared and their
+    // state set to Unloaded before next clean up
+    tile.setState(TileLoadState::Done);
```

Comment:
I don't think this is safe. Setting it to `Done` will give the selection algorithm free reign to render this tile. Which will leads to holes. Perhaps `Unloading`?